### DISCUSSION TITLE
Checking current authstore in pb_hooks?

### DISCUSSION THREAD

**Discussion by latchk3y** [pocketbase/pocketbase#4156] [Q&A] [answered]
What is the JS method of accessing the current authentication parameters of the connected Pocketbase application? 

```js
routerAdd("get", "/dashboard", (c) => {

    const info   = $apis.requestInfo(c);
    const admin  = info.admin;      
    const record = info.authRecord;

    console.log(Object.keys(info.headers));

    const isGuest = !admin && !record

    let html = "hello world!";

    if (!isGuest) {

        html = $template.loadFiles(
            `${__hooks}/views/base.html`,
            `${__hooks}/views/sidebar.html`,
            `${__hooks}/views/dash.html`
        ).render();

    } else {

        html = $template.loadFiles(
            `${__hooks}/views/base.html`,
            `${__hooks}/views/login.html`
        ).render();

    }

    return c.html(200, html);

});
```

Despite being able to use pb.authStore() to verify on the page I'm logged in, both `admin` and `record` always return null. I'm unsure if I'm meat to be doing something else.

**Comment by ganigeorgiev** [comment, marked as answer]
A request is considered authenticated if it has valid `Authorization: TOKEN` header with it.
This header is automatically added with the SDKs, but you need to send the request to your custom route with the SDK (see [Sending requests to custom routes](https://pocketbase.io/docs/js-routing/#sending-request-to-custom-routes-using-the-sdks)).

As far as I see from your code sample the above will return an html response and it is not a stateless API route.
If you are accessing the above route directly from the browser urlbar you'll need some other mechanism to send the token with because there will be no `Authorization` header with the request, for example with cookies (see also https://github.com/pocketbase/pocketbase/discussions/3447).

**Reply by latchk3y** [reply]
This information has helped me to construct something that actually works, thank you much.

**Reply by JacobJax** [reply]
How did you solve it. @latchk3y

**Reply by ganigeorgiev** [reply]
@JacobJax You are commenting on an old discussion that targets PocketBase < v0.23.

If you haven't already, please read https://pocketbase.io/docs/how-to-use/.

You'll have better chance someone to assist you if you open a new Q&A discussion and elaborate clearly on your problem and what actually you are trying to do.

**Comment by bridevmx** [comment]
If my ".html" document needs JS and CSS do I have to use $template.loadFiles() or it is not necessary,
